From 9d32df80e33a7541658858497f45bed1e59e3621 Mon Sep 17 00:00:00 2001
From: Richard Purdie <richard.purdie@linuxfoundation.org>
Date: Mon, 1 Mar 2021 13:00:47 +0000
Subject: [PATCH 06/11] qemu: Determinism fixes

When sources are included within debug information, a couple of areas of the
qemu build are not reproducible due to either full buildpaths or timestamps.

Replace the full paths with relative ones. I couldn't figure out how to get
meson to pass relative paths but we can fix that in the script.

Upstream-Status: Pending [some version of all/part of this may be accepted]
RP 2021/3/1
---
 scripts/decodetree.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/decodetree.py b/scripts/decodetree.py
index e8b72da3a9..5cd86b1428 100644
--- a/scripts/decodetree.py
+++ b/scripts/decodetree.py
@@ -1558,7 +1558,7 @@ def main():
     toppat = ExcMultiPattern(0)
 
     for filename in args:
-        input_file = filename
+        input_file = os.path.relpath(filename)
         f = open(filename, 'rt', encoding='utf-8')
         parse_file(f, toppat)
         f.close()
-- 
2.44.0

